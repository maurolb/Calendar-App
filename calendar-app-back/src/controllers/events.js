const Event = require('../models/Event');

const getAll = async () => {
  const events = await Event.find()
    .populate('user', 'name');
  return events;
};

const post = async (data, uid) => {
  const event = new Event(data);

  event.user = uid;

  const newEvent = event.save();

  return newEvent;
};

const update = async (data, uid, eventId) => {
  const event = await Event.findById({ _id: eventId });

  if (!event) {
    return 'error not-found';
  }

  if (event.user.toString() !== uid) {
    return 'error unauthorized';
  }

  const newEvent = {
    ...data,
    user: uid,
  };

  const updatedEvent = await Event.findByIdAndUpdate(eventId, newEvent, { new: true });

  return updatedEvent;
};

const remove = async (eventId, uid) => {
  const event = await Event.findById({ _id: eventId });
  if (!event) {
    return 'error not-found';
  }

  if (event.user.toString() !== uid) {
    return 'error unauthorized';
  }

  await Event.findOneAndDelete({ _id: eventId });
  return true;
};

module.exports = {
  getAll,
  post,
  update,
  remove,
};
