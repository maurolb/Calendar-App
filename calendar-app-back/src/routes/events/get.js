const { getAll } = require('../../controllers/events');

const getEvents = async (req, res) => {
  try {
    const events = await getAll();
    return res.status(200).json({ ok: true, events });
  } catch (error) {
    console.log('error ', error.message);
    return res.status(500).json({ ok: false, message: 'Error interno no se pudo obtener los eventos' });
  }
};

module.exports = getEvents;
