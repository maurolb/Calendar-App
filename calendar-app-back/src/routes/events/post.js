const { post } = require('../../controllers/events');

const postEvent = async (req, res) => {
  try {
    const newPost = await post(req.body, req.uid);
    return res.status(200).json({ ok: true, newPost });
  } catch (error) {
    console.log('error ', error.message);
    return res.status(500).json({ ok: false, message: 'Error interno no se pudo crear el evento' });
  }
};

module.exports = postEvent;
