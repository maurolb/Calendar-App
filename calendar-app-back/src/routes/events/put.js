const { update } = require('../../controllers/events');

const putEvent = async (req, res) => {
  try {
    const updatedEvent = await update(req.body, req.uid, req.params.id);
    if (updatedEvent === 'error not-found') {
      return res.status(404).json({
        ok: false,
        message: 'No se encontr√≥ id del evento',
      });
    }
    if (updatedEvent === 'error unauthorized') {
      return res.status(401).json({
        ok: false,
        message: 'No se pudo modificar, el evento no te pertenece',
      });
    }
    return res.status(200).json({ ok: true, updatedEvent });
  } catch (error) {
    console.log('error ', error.message);
    return res.status(500).json({ ok: false, message: 'Error interno, no se pudo actualizar el evento' });
  }
};

module.exports = putEvent;
