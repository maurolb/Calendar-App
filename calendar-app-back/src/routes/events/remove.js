const { remove } = require('../../controllers/events');

const removeEvent = async (req, res) => {
  try {
    const removedEvent = await remove(req.params.id, req.uid);
    if (removedEvent === 'error not-found') {
      return res.status(404).json({
        ok: false,
        message: 'No se encontr√≥ id del evento',
      });
    }

    if (removedEvent === 'error unauthorized') {
      return res.status(401).json({
        ok: false,
        message: 'No se pudo eliminar, el evento no te pertenece',
      });
    }

    return res.status(200).json({ ok: true });
  } catch (error) {
    console.log('error ', error.message);
    return res.status(500).json({ ok: false, message: 'Error interno, no se pudo borrar el evento' });
  }
};

module.exports = removeEvent;
