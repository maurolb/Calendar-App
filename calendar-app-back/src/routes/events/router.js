const { Router } = require('express');
const { check } = require('express-validator');
const isDate = require('../../helpers/isDate');
const fieldValidators = require('../../middlewares/field-validators');
const getEvents = require('./get');
const postEvent = require('./post');
const putEvent = require('./put');
const removeEvent = require('./remove');

const router = Router();

router.route('/')
  .get(
    getEvents,
  );

router.route('/')
  .post(
    [
      check('title', 'El título es requerido').not().isEmpty(),
      check('start', 'La hora de inicio es requerida').custom(isDate),
      check('end', 'La hora de finalización es requerida').custom(isDate),
      fieldValidators,
    ],
    postEvent,
  );

router.route('/:id')
  .put(
    putEvent,
  );

router.route('/:id')
  .delete(
    removeEvent,
  );

module.exports = router;
