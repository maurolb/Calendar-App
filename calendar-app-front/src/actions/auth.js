import Swal from "sweetalert2";
import { fetchConToken, fetchSinToken } from "../helpers/fetch";
import { types } from "../types/types";
import { eventLogout } from "./events";

export const startLogin = (email, password) => {
  return async (dispatch) => {
    const resp = await fetchSinToken('auth', {email, password}, 'POST');
    const body = await resp.json();
    
    if( body.ok ){
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime() );
      dispatch( login( {uid: body.uid, name: body.name} ))
    } else {
      Swal.fire('Error', body.message, 'error')
    }

  }
};

const login = ( user ) => ({
  type: types.authLogin,
  payload: user 
});

export const startRegister = (email, password, name) => {
  return async (dispatch) => {
    const resp = await fetchSinToken('auth/register', {email, password, name}, 'POST');
    const body = await resp.json();

    if( body.ok ){
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime() );
      dispatch( login( {uid: body.uid, name: body.name} ))
    } else {
      Swal.fire('Error', body.message, 'error')
    }

  }
};

export const startChecking = () => {
  return async (dispatch) => {
    const resp = await fetchConToken('auth/renew');
    const body = await resp.json();

    if( body.ok ){
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime() );

      dispatch( login( {uid: body.uid, name: body.name} ))

    } else {
      dispatch( checkingFinish() )
    }

  }
};

const checkingFinish = () => ({
  type: types.authFinishChecking
})

export const startLogout = () => {
  return async (dispatch) => {
    localStorage.clear();
    dispatch( eventLogout() );
    dispatch( logout() );
  }
}

const logout = () => ({
  type: types.authLogout
})