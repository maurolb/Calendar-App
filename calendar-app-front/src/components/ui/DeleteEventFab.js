import React from 'react'
import { useDispatch } from 'react-redux';
import { startEventDelete } from '../../actions/events';

export const DeleteEventFab = () => {

  const dispatch = useDispatch();

  const clickFabButton = () => {
    dispatch( startEventDelete() )
  }

  return (
    <button
      className="btn btn-danger delete"
      onClick={clickFabButton}
    >
      <i className="fas fa-trash"></i> Borrar Evento
    </button>
  )
}
