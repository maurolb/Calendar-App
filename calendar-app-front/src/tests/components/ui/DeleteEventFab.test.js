import { mount } from "enzyme";
import { Provider } from "react-redux";

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { startEventDelete } from "../../../actions/events";
import { DeleteEventFab } from "../../../components/ui/DeleteEventFab";

jest.mock('../../../actions/events', () => ({
  startEventDelete: jest.fn()
}));

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initialState = {}

const store = mockStore(initialState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <DeleteEventFab />
  </Provider>
)

describe('Tests on DeleteEventFab component', () => {
  
  test('should show correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });
  
  test('should call startEventDelete on click', () => {
    wrapper.find('button').simulate('click');
    
    expect(startEventDelete).toHaveBeenCalled();
  })
  
});
