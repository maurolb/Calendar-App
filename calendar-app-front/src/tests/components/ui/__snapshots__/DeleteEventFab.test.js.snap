// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on DeleteEventFab component should show correctly 1`] = `
<button
  className="btn btn-danger delete"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
   Borrar Evento
</button>
`;
